### PR TITLE
Minor fixes and CI cleanup

### LINKED ISSUE #10: Commit tagged as `v0.7.0` actually has version 0.6.0 in Cargo.toml
At time of writing, this https://github.com/aantn/smag/commit/d1d7b62d842099ba708870ec024cca14cbb3db9b is tagged as [`v0.7.0`](https://github.com/aantn/smag/blob/v0.7.0/Cargo.toml). However, the corresponding Cargo.toml has [`version = "0.6.0"`](https://github.com/aantn/smag/blob/d1d7b62d842099ba708870ec024cca14cbb3db9b/Cargo.toml#L3) in it.

### PR DESCRIPTION
**Pull request by svetter**
- Fixed version tag and repository link in `Cargo.toml` file (Closes #10)
- Extended .gitignore
- Applied all `cargo fmt` and `clippy` suggestions (`fmt` is very opinionated...)
- Replaced unmaintained actions in workflow which use deprecated mechanisms

CI runs clean and smooth now, except for a caching error which didn't seem worth going into for me since the whole CI runs pretty fast anyway.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -1,9 +1,9 @@
 [package]
 name = "smag"
-version = "0.6.0"
+version = "0.7.0"
 authors = ["Natan Yellin", "Tom Forbes <tom@tomforb.es>"]
```
Comment (aantn) [inline Cargo.toml]: Please add yourself :)

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -28,34 +29,33 @@ jobs:
...
       - name: Cache cargo index
-        uses: actions/cache@v1
+        uses: actions/cache@v4
         with:
           path: ~/.cargo/git
           key: ${{ runner.os }}-cargo-index-${{ hashFiles('**/Cargo.lock') }}
 
       - name: Run
-        uses: actions-rs/cargo@v1
+        uses: clechasseur/rs-cargo@v2
```
Comment (aantn) [inline .github/workflows/test.yml]: Is it a must to change the action? I'm particularly considered about this one as it isn't widely used. 

(The concern here is supply chain security.)

Comment (svetter) [reply .github/workflows/test.yml]: It is not a must, the CI will still run with actions-rs/cargo@v1, but GitHub will show this deprecation warning every time (e.g. see [here](https://github.com/svetter/smag/actions/runs/7913575921) in my fork before I replaced the action):

> Node.js 16 actions are deprecated. Please update the following actions to use Node.js 20: actions/cache@v1, actions-rs/cargo@v1. For more information see: https://github.blog/changelog/2023-09-22-github-actions-transitioning-from-node-16-to-node-20/.

Another possibility would be to use no action at all but just the run command:

```yaml
- name: run
  run: |
    cargo run -- --help
```

As I understand it, this was not possible in the past, probably including the time when gping was written, but it is now (although I haven't tested it myself).

Let me know what you want to go with. We can also just revert the change, of course.

File: Cargo.toml (hunk 1 above)
Comment (svetter) [reply Cargo.toml]: Thanks! 🙂

File: .github/workflows/test.yml (hunk 2 above)
Comment (aantn) [reply .github/workflows/test.yml]: If we can still run with the old action, lets do so. I'd love to get this merged and a new version released, so lets handle updating the actions another time.

Comment (svetter) [reply .github/workflows/test.yml]: What about the other replaced/updated actions?

**Comment by aantn** [comment]
If it's possible to keep the older ones there too, let's do it for now.

**Comment by svetter** [comment]
Action changes are removed.

**Review by aantn — APPROVED** [review]

**Comment by svetter** [comment]
@aantn Reminder that you said you wanted to create a new release ;-)

**Comment by aantn** [comment]
Done :)

**Pull request merged**